IDEAS/Examples/IBPSA/SingleZoneResidentialHydronicHeatPump.mo

```diff
@@ -0,0 +1,709 @@
+        TSetHeaOcc else TSetHeaUno) "Heating temperature setpoint with setback"
+    annotation (Placement(transformation(extent={{-220,-40},{-200,-20}})));
+  Utilities.IO.SignalExchange.Read reaHeaPumY(description="Block for reading the heat pump modulating signal",
+      y(unit="1")) "Read heat pump modulating signal"
+    annotation (Placement(transformation(extent={{120,140},{140,160}})));
+  Utilities.IO.SignalExchange.Read reaPum(description=
+        "Control signal for emission cirquit pump", y(unit="1"))
+    "Read control signal for emission circuit pump"
+    annotation (Placement(transformation(extent={{22,100},{42,120}})));
+  Modelica.Blocks.Continuous.LimPID conPI(
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    k=0.6,
+    Ti=8000,
+    yMax=1,
+    yMin=0,
+    initType=Modelica.Blocks.Types.InitPID.InitialState)
+    "PI controller for the boiler supply water temperature"
+    annotation (Placement(transformation(extent={{20,140},{40,160}})));
+  Modelica.Blocks.Math.Add addOcc
+    annotation (Placement(transformation(extent={{-160,120},{-140,140}})));
+  Fluid.Movers.FlowControlled_dp pum(
+    inputType=IDEAS.Fluid.Types.InputType.Stages,
+    massDynamics=Modelica.Fluid.Types.Dynamics.SteadyState,
+    use_inputFilter=false,
+    dp_nominal=20000,
+    m_flow_nominal=0.5,
+    redeclare package Medium = MediumWater,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState)
+    "Circulation pump for emission system"
+    annotation (Placement(transformation(extent={{40,30},{20,50}})));
+  Fluid.Sensors.TemperatureTwoPort senTemSup(
+    redeclare package Medium = MediumWater,
+    m_flow_nominal=pum.m_flow_nominal,
+    tau=0) "Supply water temperature sensor"
+    annotation (Placement(transformation(extent={{80,50},{60,30}})));
+  Fluid.HeatExchangers.RadiantSlab.EmbeddedPipe floHea(
+    redeclare package Medium = MediumWater,
+    redeclare Fluid.HeatExchangers.RadiantSlab.BaseClasses.FH_Standard1
+      RadSlaCha,
+    allowFlowReversal=true,
+    m_flow_nominal=pum.m_flow_nominal,
+    dp_nominal=pum.dp_nominal/2,
+    A_floor=case900Template.AZone) "Floor heating of the zone"
+    annotation (Placement(transformation(extent={{0,0},{-20,20}})));
+  Fluid.Sensors.TemperatureTwoPort senTemRet(
+    redeclare package Medium = MediumWater,
+    m_flow_nominal=pum.m_flow_nominal,
+    tau=0) "Return water temperature sensor"
+    annotation (Placement(transformation(extent={{80,-10},{60,-30}})));
+  Fluid.HeatPumps.ScrollWaterToWater heaPum(
+    redeclare package Medium1 = MediumWater,
+    redeclare package Medium2 = MediumAir,
+    scaling_factor=1,
+    m2_flow_nominal=fan.m_flow_nominal,
+    enable_variable_speed=true,
+    m1_flow_nominal=pum.m_flow_nominal,
+    T1_start=293.15,
+    T2_start=278.15,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    TEvaMin=253.15,
+    dTHys=3,
+    dp1_nominal=pum.dp_nominal/2,
+    dp2_nominal=fan.dp_nominal/2,
```

Comment:
why divide by two?

Reply:
In evaporator circuit it makes no sense and has been removed in https://github.com/open-ideas/IDEAS/pull/1151/commits/45f687f26d2b2bc6b899ca347fab14931276ff52, it was originally coming from the previous setup that was using an air/water heat exchanger. For the water circuit it's just to distribute the nominal pressure head of the pump among the floor heating and the heat pump.